servers/zms/src/main/java/com/yahoo/athenz/zms/QuotaChecker.java

```diff
@@ -378,4 +379,44 @@ void checkEntityQuota(ObjectStoreConnection con, String domainName, Entity entit
                     + quota.getEntity() + " actual: " + objectCount, caller);
         }
     }
+    void checkAssertionConditionsQuota(ObjectStoreConnection con, String domainName, long assertionId, AssertionConditions assertionConditions,
+                                       String caller) {
+
+        // if quota check is disabled we have nothing to do
+        if (!quotaCheckEnabled) {
+            return;
+        }
+
+        // if our assertionConditions is null then there is no quota check
+        if (assertionConditions == null || assertionConditions.getConditionsList() == null ||
+                assertionConditions.getConditionsList().isEmpty()) {
+            return;
+        }
+
+        // we're going to check if we'll be allowed to create given assertionConditions
+        countAssertionConditions(con, assertionId, assertionConditions.getConditionsList().size(), caller);
```

Comment:
in the method assertionConditions.getConditionsList().size() is treated as newCount - but is it the new count? If i'm updating the condition, does the method correctly handle that there are no new conditions so it's fine?